internal/app/app.go

```diff
@@ -0,0 +1,153 @@
+// Package app exports a D-Bus object to apply proxy settings.
+package app
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/godbus/dbus/v5"
+	"github.com/godbus/dbus/v5/introspect"
+	log "github.com/sirupsen/logrus"
+	"github.com/ubuntu/decorate"
+	"github.com/ubuntu/ubuntu-proxy-manager/internal/proxy"
+)
+
+const dbusObjectPath = "/com/ubuntu/ProxyManager"
+const dbusInterface = "com.ubuntu.ProxyManager"
+
+// DryRun is a context key to indicate that we should not apply the proxy
+// settings. It is intended to be used in tests only.
+type DryRun struct{}
```

Comment:
Actually, to trim it down even more, it’s better to really instantiate an object and use it, like https://github.com/ubuntu/adsys/blob/main/internal/grpc/logstreamer/server.go#L17.